packages/eslint-plugin-sui/src/rules/factory-pattern.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * @fileoverview Ensure that our classes are using the convetion of has a static create method as factory.
+ * @author factory pattern
+ */
+'use strict'
+
+const dedent = require('string-dedent')
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+/** @type {import('eslint').Rule.RuleModule} */
+module.exports = {
+  meta: {
+    type: 'warning',
+    docs: {
+      description: 'ensure to define at least one factory function',
+      recommended: true,
+      url: 'https://github.mpi-internal.com/scmspain/es-td-agreements/blob/master/30-Frontend/00-agreements/02-project-structure.md'
```

Comment:
I'd put the link to the actual agreement. If we don't have it yet, then I'd link it to the agreements index: https://github.mpi-internal.com/scmspain/es-td-agreements/blob/master/30-Frontend/00-agreements

Reply:
👍